ql/src/test/results/clientpositive/llap/order_null.q.out

```diff
@@ -116,12 +116,12 @@ POSTHOOK: query: SELECT x.* FROM src_null_n1 x ORDER BY b desc, a asc
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@src_null_n1
 #### A masked pattern was here ####
-2	B
-1	A
-2	A
 2	NULL
 3	NULL
 NULL	NULL
+2	B
+1	A
+2	A
```

Comment:
The result that I would expect is:
```
2         B
1         A
2         A
NULL  NULL
2         NULL
3         NULL
```

Reply:
Not sure about how up-to-date that document is but as far as I know we want nulls last as the default for asc and nulls first for desc.
It can be controlled by the setting `hive.default.nulls.last`. Default value is `true`.

I also run these queries with Postgres and got the same results as with this patch.

Reply:
OK, now I understand better, thanks for the clarification.

The `hive.default.nulls.last` is a bit confusing. From the name and description I get the impression that NULLS LAST is the default behavior for both ASC and DESC and apparently it is not the case either. 

Yes the new results are in sync with Postgres since the latter uses the [semantics that you mentioned](https://www.postgresql.org/docs/12/queries-order.html). Note that this is not a rule from every DBMS.

Reply:
I agree that description is confusing I changed accordingly.